src/integrations/ga/GaToSplit.js

```diff
@@ -1,7 +1,257 @@
+    // We set default identities if key and TT are present in settings.core
+    identities: (coreSettings.key && coreSettings.trafficType) ?
+      [{ key: coreSettings.key, trafficType: coreSettings.trafficType }] :
+      undefined
+  };
+
+  class SplitTracker {
+
+    // Constructor for the SplitTracker plugin.
+    constructor(tracker, pluginOptions) {
+
+      // precedence of options: SDK options (config.integrations) overwrite pluginOptions (`ga('require', 'splitTracker', pluginOptions)`)
+      const opts = Object.assign({}, defaultOptions, sdkOptions, pluginOptions);
+
+      this.tracker = tracker;
+
+      // Validate identities
+      const validIdentities = validateIdentities(opts.identities);
+
+      if (!validIdentities || validIdentities.length === 0) {
+        log.warn('No valid identities were provided. Please check that you are passing a valid list of identities or providing a traffic type at the SDK configuration.');
+        return;
+      }
+
+      const invalids = validIdentities.length - opts.identities.length;
+      if (invalids) {
+        log.warn(`${invalids} identities were discarded because they are invalid or duplicated. Identities must be an array of objects with key and trafficType.`);
+      }
+      opts.identities = validIdentities;
+
+      // Validate prefix
+      // @TODO Improve the prefix validation using the same REGEX than eventTypeId
+      if (opts.prefix && !isString(opts.prefix)) {
+        log.warn('The provided `prefix` was ignored since it is invalid. Please check that you are passing a string object as `prefix`.');
+        opts.prefix = undefined;
+      }
+
+      // Overwrite sendHitTask to perform plugin tasks:
+      // 1) filter hits
+      // 2) map hits to Split events
+      // 3) handle events, i.e., validate and send them to Split BE
+      const originalSendHitTask = tracker.get('sendHitTask');
+      tracker.set('sendHitTask', function (model) {
+        originalSendHitTask(model);
+
+        // filter and map hits into an EventData instance
+        const eventData = opts.filter(model) && opts.mapper(model);
+
+        // don't send the event if it is falsy or invalid when generated by a custom mapper
+        if (!eventData || (opts.mapper !== defaultMapper && !validateEventData(eventData)))
+          return;
+
+        // Add prefix
+        if (opts.prefix) eventData.eventTypeId = opts.prefix + eventData.eventTypeId;
+
+        // Add timestamp if not present
+        if (!eventData.timestamp) eventData.timestamp = Date.now();
+
+        // Store the event
+        if (eventData.key && eventData.trafficTypeName) {
+          opts.eventHandler(event);
+        } else { // Store the event for each Key-TT pair (identities), if key and TT is not present in eventData
+          opts.identities.forEach(identity => {
+            const event = Object.assign({
+              key: identity.key,
+              trafficTypeName: identity.trafficType,
+            }, eventData);
```

Comment:
If the event data were to contain only one of those values set, that would override the identity one. I guess that's expected right?

Reply:
Yes